pallet/fee-proxy/src/impls.rs

```diff
@@ -182,3 +220,131 @@ where
+/// can be invalid in the following cases:
+///   - The sylo call has been wrapped in a call_with_fee_preferences call. Sylo
+///     calls should be paid in Sylos only.
+///   - The sylo call is in a batch/batch_all call. In batch calls, if any call is
+///     a sylo call, then all inner calls must be a sylo call. This simplifies the
+///     implementation, preventing a need to process the fee for each individual call.
+fn is_sylo_and_valid_call<T>(
+	call: &<T as frame_system::Config>::RuntimeCall,
+) -> Result<bool, TransactionValidityError>
+where
+	T: Config
+		+ frame_system::Config<AccountId = AccountId>
+		+ pallet_futurepass::Config
+		+ pallet_xrpl::Config
+		+ pallet_proxy::Config
+		+ pallet_utility::Config
+		+ pallet_sylo_data_verification::Config,
+	<T as frame_system::Config>::RuntimeCall: IsSubType<crate::Call<T>>,
+	<T as frame_system::Config>::RuntimeCall: IsSubType<pallet_futurepass::Call<T>>,
+	<T as frame_system::Config>::RuntimeCall: IsSubType<pallet_sylo_data_verification::Call<T>>,
+	<T as frame_system::Config>::RuntimeCall: IsSubType<pallet_proxy::Call<T>>,
+	<T as frame_system::Config>::RuntimeCall: IsSubType<pallet_utility::Call<T>>,
+	<T as frame_system::Config>::RuntimeCall: IsSubType<pallet_xrpl::Call<T>>,
+	<T as Config>::RuntimeCall: IsSubType<pallet_futurepass::Call<T>>,
+	<T as Config>::RuntimeCall: IsSubType<pallet_xrpl::Call<T>>,
+	<T as Config>::RuntimeCall: IsSubType<pallet_sylo_data_verification::Call<T>>,
+	<T as Config>::RuntimeCall: IsSubType<pallet_utility::Call<T>>,
+	<T as pallet_futurepass::Config>::RuntimeCall:
+		IsSubType<pallet_sylo_data_verification::Call<T>>,
+{
+	if match call.is_sub_type() {
+		Some(pallet_sylo_data_verification::Call::register_resolver { .. }) => true,
+		Some(pallet_sylo_data_verification::Call::update_resolver { .. }) => true,
+		Some(pallet_sylo_data_verification::Call::deregister_resolver { .. }) => true,
+		Some(pallet_sylo_data_verification::Call::create_validation_record { .. }) => true,
+		Some(pallet_sylo_data_verification::Call::add_validation_record_entry { .. }) => true,
+		Some(pallet_sylo_data_verification::Call::update_validation_record { .. }) => true,
+		Some(pallet_sylo_data_verification::Call::delete_validation_record { .. }) => true,
+		_ => false,
+	} {
+		return Ok(true);
+	}
+
+	// check if the inner call of a futurepass call is a sylo call
+	if let Some(pallet_futurepass::Call::proxy_extrinsic { call, .. }) = call.is_sub_type() {
+		return is_sylo_and_valid_call::<T>(call.as_ref().into_ref());
+	}
+
+	// check if the inner call of a proxy pallet call is a sylo call
+	let is_proxy = match call.is_sub_type() {
+		Some(pallet_proxy::Call::proxy { call, .. }) => {
+			is_sylo_and_valid_call::<T>(call.as_ref().into_ref())
+		},
+		Some(pallet_proxy::Call::proxy_announced { call, .. }) => {
+			is_sylo_and_valid_call::<T>(call.as_ref().into_ref())
+		},
+		_ => Ok(false),
+	}?;
+
+	if is_proxy {
+		return Ok(true);
+	}
```

Comment:
nit: can remove `is_proxy` check
```suggestion
	let is_proxy = match call.is_sub_type() {
		Some(pallet_proxy::Call::proxy { call, .. }) => {
			return is_sylo_and_valid_call::<T>(call.as_ref().into_ref());
		},
		Some(pallet_proxy::Call::proxy_announced { call, .. }) => {
			return is_sylo_and_valid_call::<T>(call.as_ref().into_ref());
		},
		_ => Ok(false),
	}?;
```